finality-aleph/src/network.rs

```diff
@@ -87,8 +75,11 @@ pub trait Network<B: BlockT>: Clone + Send + Sync + 'static {
     /// TODO: figure out what does this actually do...
     fn remove_set_reserved(&self, who: PeerId, protocol: Cow<'static, str>);
 
-    // The PeerId of this node.
+    /// The PeerId of this node.
     fn peer_id(&self) -> PeerId;
+
+    /// Request the justification for the given block
+    fn request_justification(&self, hash: &B::Hash, number: NumberFor<B>);
```

Comment:
This seems like a strange place to require this function. Cannot we send it as one of the enum variants in `DataNetwork`? Or does this work in a strange way that is very entangled with substrate?

Reply:
The `Network` trait is something that should be discussed in more detail. There are unused methods and methods which we don't know what are supposed to do?

Reply:
This is part of some very specific API that substrate's NetworkService provides. There are probably alternatives on how to do that, but I don't see how to remove it from this trait.

Reply:
Well, the only interface needed by `ConsensusNetwork` are the methods
* `async fn next_event(&self) -> Event<D>` where `Event<D>` is an enum with variants `PeerConnected(PeerId)`, `PeerDisconnected(PeerId)` and `IncomingMessage(InternalMessage<D>)`,
* `fn send_message(&self, peer_id: PeerId, msg: InternalMessage<D>)`,
* and maybe `fn request_justification(&self, hash: &B::Hash, number: NumberFor<B>)`, but this can be just another variant `InternalMessage` which is handled specially in the implementation.

This require writing a (slightly) non-trivial implementation of the trait, and maybe even having an adapter, but this seems more reasonable than having a trait which precisely reflects a substrate struct

Reply:
I'm gonna assume this will be dealt with as part of the network refactor task and I'm resolving this.